src/realm/object-store/sync/sync_session.hpp

```diff
@@ -367,8 +368,9 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     _impl::SyncProgressNotifier m_progress_notifier;
     ConnectionChangeNotifier m_connection_change_notifier;
 
+    mutable util::CheckedMutex m_external_reference_mutex;
```

Comment:
this seems like a good time to add static checking around m_state_mutex as well?

Reply:
It turned out to be a bigger task that I don't think belongs in this PR.